src/Nox.Types.EntityFramework/Extensions/EntityRelationshipExtensions.cs

```diff
@@ -35,9 +35,15 @@ public static bool ConfigureThisSide(this EntityRelationship relationship)
         // Single to Single pick Exactly one
         if (relationship.WithSingleEntity && reverseRelationship.WithSingleEntity)
         {
-           return relationship.Relationship == EntityRelationshipType.ExactlyOne;
+            return relationship.Relationship == EntityRelationshipType.ExactlyOne;
         }
-        
+
         throw new NotSupportedException($"Can not define Relationship side to configure in EntityFramework {relationship.Name}");
     }
+
+    public static string? GetNavigationPropertyName(this EntityRelationship thisSide, EntityRelationship otherSize)
+        => IsSelfReferencingRelationshipTo(thisSide, otherSize) ? null : thisSide.Related.Entity.GetNavigationPropertyName(otherSize);
+
+    public static bool IsSelfReferencingRelationshipTo(this EntityRelationship thisSide, EntityRelationship otherSize)
```

Comment:
small sugestion to improve readbility
```suggestion
    public static bool IsToItself(this EntityRelationship thisSide, EntityRelationship otherSize)
```